networkx/algorithms/dag.py

```diff
@@ -79,172 +82,184 @@ def is_directed_acyclic_graph(G):
     if not G.is_directed():
         return False
     try:
-        topological_sort(G, reverse=True)
+        consume(topological_sort(G))
```

Comment:
Because topological_sort is now a generator function, it will not raise that exception until the iterator is consumed.